README.md

```diff
@@ -9,17 +9,39 @@ This repository contains a collection of engines that power the core stack for
 Client](https://www.prisma.io/client) and [Prisma
 Migrate](https://www.prisma.io/migrate).
 
-The engines and their respective binary crates are:
+If you're looking for how to install Prisma or any of the engines, the [Getting
+Started](https://www.prisma.io/docs/getting-started) guide might be useful.
 
-- Query engine: `query-engine`
-- Migration engine: `migration-engine-cli`
-- Introspection engine: `introspection-engine`
-- Prisma Format: `prisma-fmt`
+This document describes some of the internals of the engines, and how to build
+and test them.
+
+## What's in this repository
+
+This repository contains four binaries:
+
+- *Query engine*, used by the client to run database queries from Prisma Client
+- *Migration engine*, used to create and run migrations
+- *Introspection engine*, used to get the state of the database, compare it
+    to the schema file, and figure out what the differences are
+- *Prisma Format*, used to format prisma files
+
+Additionally, the *psl* (Prisma Schema Language) is the library that defines how
+the language looks like, how it's parsed, etc.
+
+You'll also find:
+- *libs*, for various (small) libraries such as macros, user facing errors,
+    various connector/database-specific libraries, etc.
+- a `docker-compose.yml` file that's helpful for running tests and bringing up
+    containers for various databases
+- a `flake.nix` file for bringing up all dependencies and making it easy to
+    build the code in this repository
```

Comment:
Nitpick: I think we should mention somewhere that the nix setup is there for people who use nix or want a reliable, executable, reproducible description of how to build the engines, but it can be completely ignored. On the other hand this bullet point is not making it sound like a requirement so I think it can stay as it is.

Reply:
Good point, I clarified!